src/RestWrite.js

```diff
@@ -478,106 +448,157 @@ RestWrite.prototype.filteredObjectsByACL = function (objects) {
   });
 };
 
-RestWrite.prototype.handleAuthData = function (authData) {
-  let results;
-  return this.findUsersWithAuthData(authData).then(async r => {
-    results = this.filteredObjectsByACL(r);
-
-    if (results.length == 1) {
-      this.storage['authProvider'] = Object.keys(authData).join(',');
-
-      const userResult = results[0];
-      const mutatedAuthData = {};
-      Object.keys(authData).forEach(provider => {
-        const providerData = authData[provider];
-        const userAuthData = userResult.authData[provider];
-        if (!_.isEqual(providerData, userAuthData)) {
-          mutatedAuthData[provider] = providerData;
-        }
-      });
-      const hasMutatedAuthData = Object.keys(mutatedAuthData).length !== 0;
-      let userId;
-      if (this.query && this.query.objectId) {
-        userId = this.query.objectId;
-      } else if (this.auth && this.auth.user && this.auth.user.id) {
-        userId = this.auth.user.id;
+RestWrite.prototype.getUserId = function () {
+  if (this.query && this.query.objectId && this.className === '_User') {
+    return this.query.objectId;
+  } else if (this.auth && this.auth.user && this.auth.user.id) {
+    return this.auth.user.id;
+  }
+};
+
+// Developers are allowed to change authData via before save trigger
+// we need after before save to ensure that the developer
+// is not currently duplicating auth data ID
+RestWrite.prototype.ensureUniqueAuthDataId = async function () {
+  if (this.className !== '_User') return;
+  if (!this.data.authData) return;
```

Comment:
Single line return

Reply:
Agree!

Reply:
@dblythy  not sur to understand the "single line" comment. You want a single "if" or i need to add brackets ?

Reply:
I think the preferred styling should be to always have curly brackets whenever there’s an if statement